### PR TITLE
fix: show correct staking balance by account id

### LINKED ISSUE #2614: Cosmos asset/account details page
### Overview

This page is a "plugin" and looks identical but contains different logic

### References and additional details

* start here https://github.com/shapeshift/web/blob/develop/src/plugins/cosmos/CosmosAssetAccountDetails.tsx#L21

### Acceptance Criteria

* staking section must support multiple accounts
* asset allocation section must support multiple accounts
* chart for asset page must show aggregated staked balance across all accounts, including delegations/redelegations/undelegations
* chart for account page must show account balance as per above

### Need By Date

_No response_

### Screenshots/Mockups

image

### Estimated effort

2 days

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

previously we were aggregating all staked balance by asset id, this adds a selector and filters by both asset and account id's

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

related to but does not close #2614 

## Risk

* we show incorrect staked amounts above the chart
* we show incorrect cosmossdk account balances on the accounts page

## Testing

* turn multi account flag on
* add a cosmos account
* navigate to new cosmos account (can have balance or not)
* new cosmos account should not show any staked balance from account 0. if new account has a staked balance it should show that amount correctly.

### Engineering

* see common testing steps

### Operations

* see common testing steps

## Screenshots (if applicable)

before (account with a single tx and no staked balance)

image

after with same account

image

### CODE REVIEW

File: .env.dev [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # feature flags
 REACT_APP_FEATURE_OSMOSIS=true
-REACT_APP_FEATURE_PENDO=true
+REACT_APP_FEATURE_PENDO=false
```
Comment (0xdef1cafe) [inline .env.dev:3]: this was pissing me off locally

File: src/state/slices/portfolioSlice/selectors.ts [hunk 2]
```diff
@@ -769,27 +769,77 @@ export const selectPortfolioAccountIdsSortedFiat = createDeepEqualOutputSelector
...
-        Object.values(stakingDataByAssetId).forEach(stakingData => {
-          const { delegations, redelegations, undelegations } = stakingData
-          const redelegationEntries = redelegations.flatMap(redelegation => redelegation.entries)
-          const combined = [...delegations, ...redelegationEntries, ...undelegations]
-          combined.forEach(({ assetId, amount }) => {
-            acc[accountId][assetId] = bnOrZero(acc[accountId][assetId]).plus(amount).toString()
-          })
+  selectPortfolioStakingCryptoBalances,
+  (accountBalances, stakingBalances): PortfolioAccountBalancesById => {
+    return Object.entries(accountBalances).reduce<PortfolioAccountBalancesById>(
```
Comment (0xdef1cafe) [inline src/state/slices/portfolioSlice/selectors.ts:831]: i moved the guts of this logic up to `selectPortfolioStakingCryptoBalances`, now this selector just does a deep merge

Comment (0xdef1cafe) [reply src/state/slices/portfolioSlice/selectors.ts:831]: by deep merge i mean deep sum merge

File: src/components/AssetHeader/AssetChart.tsx [hunk 3]
```diff
@@ -72,11 +72,9 @@ export const AssetChart = ({ accountId, assetId, isLoaded }: AssetChartProps) =>
   const fiatBalance = useAppSelector(s => selectFiatBalanceIncludingStakingByFilter(s, filter))
   const cryptoBalance = useAppSelector(s => selectCryptoBalanceIncludingStakingByFilter(s, filter))
 
-  const earnBalances = useEarnBalances()
```
Comment (gomesalexandre) [inline src/components/AssetHeader/AssetChart.tsx:75]: 💜

**Review by gomesalexandre — APPROVED** [review]
LGTM, this both adds per-accountId staking balance discrimination as well as fixing the current erroneous staking balance on account 0.

Tested:

#### Account 1 - smallish delegations/undelegations balance

- This diff

image

- Prod - showing the staking balance of account 0

image

#### Account 2 - no staking balance

- This diff

image

- Prod - showing the staking balance of account 0

image

#### Account 0 for reference

- This diff (the right staked value is displayed)

image

- Prod (note, staked value is actually wrong there, as confirmed on Mintscan)

image

🧆

**Pull request merged**